sklearn/gaussian_process/_gpr.py

```diff
@@ -470,29 +481,35 @@ def log_marginal_likelihood(self, theta=None, eval_gradient=False,
         else:
             K = kernel(self.X_train_)
 
+        # Alg. 2.1, page 19, line 2 -> L = cholesky(K + sigma^2 I)
         K[np.diag_indices_from(K)] += self.alpha
         try:
-            L = cholesky(K, lower=True)  # Line 2
+            L = cholesky(K, lower=GPR_CHOLESKY_LOWER)
         except np.linalg.LinAlgError:
-            return (-np.inf, np.zeros_like(theta)) \
-                if eval_gradient else -np.inf
+            return (
+                (-np.inf, np.zeros_like(theta)) if eval_gradient else -np.inf
+            )
 
         # Support multi-dimensional output of self.y_train_
         y_train = self.y_train_
         if y_train.ndim == 1:
             y_train = y_train[:, np.newaxis]
 
-        alpha = cho_solve((L, True), y_train)  # Line 3
+        # Alg 2.1, page 19, line 3 -> alpha = L^T \ (L \ y)
+        alpha = cho_solve((L, GPR_CHOLESKY_LOWER), y_train)
 
-        # Compute log-likelihood (compare line 7)
+        # Alg 2.1, page 19, line 7
+        # -0.5 . y^T . alpha - sum(log(diag(L))) - n_samples / 2 log(2*pi)
         log_likelihood_dims = -0.5 * np.einsum("ik,ik->k", y_train, alpha)
```

Comment:
As discussed with @glemaitre, this should be:
```suggestion
        log_likelihood_dims = -0.5 * np.einsum("ik,ik->i", y_train, alpha)
```

Reply:
Actually we double checked with guillaume and expected is diag(y_train.T @ alpha) so the actual code is correct

Reply:
Great, I guess this PR can be merged then. 🙂